src/js/helpers.js

```diff
@@ -177,6 +177,39 @@ function assignments (node) {
     return tr;
 }
 
+/**
+ * Return saved grades for specified username.
+ * @param {String} username users full name
+ * @returns {Course[]} list of courses objects for that user
+ */
+function getSavedGrades (username) {
+    const courses = [];
+    let course_list = {};
+    browser.storage.local.get(["user_list"], function (data) {
+        course_list = data.user_list[username].courses;
+    });
+    for (let i = 0; i < course_list.length; i++) {
+        courses.push(new Course(course_list[i].name, course_list[i].link, course_list[i].grade, course_list[i].finalPercent, course_list[i].assignments));
+    }
+    return courses;
+}
```

Comment:
There's a race condition here if the api does not return fast enough.
Also, the WebExtension API is promised based rather then callback based. I'd recommend turning this whole function into an async function then using await for the storage call.

Also, I just learned about the `@async` tag in JSDoc. Nice!
```suggestion
/**
 * Return saved grades for specified username.
 * @async
 * @param {String} username users full name
 * @returns {Promise<Course[]>} list of courses objects for that user
 */
async function getSavedGrades (username) {
    const courses = [];
    let course_list = (await browser.storage.local.get(["user_list"]))?.user_list?.[username]?.courses || [];
    for (let i = 0; i < course_list.length; i++) {
        courses.push(new Course(course_list[i].name, course_list[i].link, course_list[i].grade, course_list[i].finalPercent, course_list[i].assignments));
    }
    return courses;
}
```

Reply:
Does the saveGradesLocally function also need to be made async since it's not returning anything?

Reply:
Good point, couldn't hurt.